DatadogCore/Sources/Core/Storage/FeatureStorage.swift

```diff
@@ -120,25 +120,37 @@ extension FeatureStorage {
         encryption: DataEncryption?,
         telemetry: Telemetry
     ) {
+        let trackName = BatchMetric.trackValue(for: featureName)
+
+        if trackName == nil {
+            DD.logger.error("Can't determine track name for feature named '\(featureName)'")
+        }
+
         let authorizedFilesOrchestrator = FilesOrchestrator(
             directory: directories.authorized,
             performance: performance,
             dateProvider: dateProvider,
             telemetry: telemetry,
-            metricsData: {
-                guard let trackName = BatchMetric.trackValue(for: featureName) else {
-                    DD.logger.error("Can't determine track name for feature named '\(featureName)'")
-                    return nil
-                }
-                return FilesOrchestrator.MetricsData(trackName: trackName, uploaderPerformance: performance)
-            }()
+            metricsData: trackName.map { trackName in
```

Comment:
> [!NOTE]
> Why are we using a `.map` operation here? it seems the `trackName` is a String, and not a list, so it seems not necessary

Reply:
It's [`Optional.map(_:)`](https://developer.apple.com/documentation/swift/optional/map(_:)), not [`Array.map(_:)`](https://developer.apple.com/documentation/swift/sequence/map(_:)) - a Swift counterpart of Kotlin's `?.let` I think.

Reply:
Oh got it ;)